File: app/src/main/java/vn/loitp/app/activity/customviews/layout/draggablepanelfree/DraggablePanelFreeActivity.java
package vn.loitp.app.activity.customviews.layout.draggablepanelfree;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.core.base.BaseFontActivity;

import loitp.basemaster.R;
import vn.loitp.views.layout.draggablepanelfree.DraggablePanelFreeLayout;

public class DraggablePanelFreeActivity extends BaseFontActivity {
    private DraggablePanelFreeLayout dpfl;
    private TextView tvState;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        dpfl = (DraggablePanelFreeLayout) findViewById(R.id.dpfl);
        tvState = (TextView) findViewById(R.id.tv_state);
        findViewById(R.id.bt_maximize).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                dpfl.maximize();
            }
        });
        findViewById(R.id.bt_minimize).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                dpfl.minimize();
            }
        });
        dpfl.setCallback(new DraggablePanelFreeLayout.Callback() {
            @Override
            public void onStateChange(DraggablePanelFreeLayout.State state) {
                tvState.setText("onStateChange " + state.name());
            }
        });
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_draggable_panel_free;
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/layout/draggablepanelfree/DraggablePanelFreeActivity.kt
package vn.loitp.app.activity.customviews.layout.draggablepanelfree

import android.annotation.SuppressLint
import android.os.Bundle
import androidx.core.view.isVisible
import com.annotation.IsFullScreen
import com.annotation.LogTag
import com.core.base.BaseFontActivity
import com.core.utilities.LUIUtil
import com.views.setSafeOnClickListener
import kotlinx.android.synthetic.main.activity_draggable_panel_free.*
import vn.loitp.app.R

@LogTag("DraggablePanelFreeActivity")
@IsFullScreen(false)
class DraggablePanelFreeActivity : BaseFontActivity() {

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_draggable_panel_free
    }

    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setupViews()
    }

    @SuppressLint("SetTextI18n")
    private fun setupViews() {
        lActionBar.apply {
            LUIUtil.setSafeOnClickListenerElastic(
                view = this.ivIconLeft,
                runnable = {
                    onBackPressed()
                }
            )
            this.ivIconRight?.isVisible = false
            this.viewShadow?.isVisible = true
            this.tvTitle?.text = DraggablePanelFreeActivity::class.java.simpleName
        }
        btMaximize.setSafeOnClickListener {
            dpfl.maximize()
        }
        btMinimize.setSafeOnClickListener {
            dpfl.minimize()
        }
        dpfl?.setCallback { state ->
            tvState.text = "onStateChange " + state.name
        }
    }
}
